lib/Models/Terria.ts

```diff
@@ -481,6 +482,12 @@ export default class Terria {
   @observable
   mapInteractionModeStack: MapInteractionMode[] = [];
 
+  @observable isWorkflowPanelActive = false;
+
+  /** Gets or sets the active SelectableDimensionWorkflow, if defined, then the workflow will be displayed using `WorkflowPanel` */
+  @observable
+  selectableDimensionWorkflow?: SelectableDimensionWorkflow;
+
```

Comment:
Suggestion: I was thinking about how an external plugin component might launch a new workflow. It could set this observable directly but that seems like reaching into internal Terria state. Another option might be to declare `SelecatableDimensionsWorkflow` also as a namespace (just like `ViewingControls`) and add a function to the namespace called `runWorkflow(viewState: ViewState, workflow: ...)` or something. That way a plugin could call `SelectableDimensionsWorkflow.runWorkflow(...)`. We could also migrate this setting to `ViewState` which seems more appropriate.

Reply:
Yep that sounds good too!
I'm not attached to this being in `Terria.ts` - I just put it in the easiest place for the moment 🙂